libraries/AP_InertialSensor/AP_InertialSensor_Backend.h

```diff
@@ -91,6 +91,12 @@ class AP_InertialSensor_Backend
     bool has_been_killed(uint8_t instance) const { return false; }
 #endif
 
+    // get the backend update rate for the gyro in Hz
+    // return 0 if the backend polling rate is the same as the sample rate or higher
+    // override return the backend rate in Hz if it is lower than the sample rate
+    virtual uint16_t get_gyro_backend_rate_hz() const {
+        return 0;
```

Comment:
This is really dangerous. The rate is quite useful, but the fact that it might be 0 could lead to all kinds of issues if it gets used for other things. I think you should return the actual rate here (raw rate?) which will make the pre-arm check simpler as well.

Reply:
Silly decision making on my end actually. This should simply return sample rate.